src/lib.rs

```diff
@@ -4,24 +4,26 @@ pub fn hex_to_base64<T: ?Sized + AsRef<[u8]>>(input: &T) -> String {
 
     let mut output = String::new();
 
-    let byte_chunks = input.as_ref().chunks(3);
+    let mut bytes = input.as_ref().to_owned();
+    let byte_chunks = bytes.chunks_mut(3);
```

Comment:
to_owned() gives us a vector of bytes, which we can make mutable.  We don't actually need to mutate the input vector, but since we *are* mutating `empty_chunk_buffer`, the type has to match.  You might also be able to do this by changing your AsRef trait bound to AsMut, and changing L7 to `input.as_mut().chunks_mut().